runtime/gc_glue_java/ObjectModel.cpp

```diff
@@ -120,7 +136,25 @@ GC_ObjectModel::internalClassLoadHook(J9HookInterface** hook, UDATA eventNum, vo
 			objectModel->_classClass = clazz;
 		} else if (0 == compareUTF8Length(J9UTF8_DATA(className), J9UTF8_LENGTH(className), (U_8*)abstractOwnableSynchronizer, strlen(abstractOwnableSynchronizer))) {
 			 clazz->classDepthAndFlags |= J9AccClassOwnableSynchronizer;
-		}
+		} else if (0 == compareUTF8Length(J9UTF8_DATA(className), J9UTF8_LENGTH(className), (U_8*)javaLangString, strlen(javaLangString))) {
```

Comment:
Another way to write this that will result in fewer comparisons is to check the length up front and switch on that to find the relevant strings to test:

```C
const uintptr_t length = J9UTF8_LENGTH(className); 
switch(length) {
case 16:
	if (0 == compareUTF8Length(J9UTF8_DATA(className), J9UTF8_LENGTH(className), (U_8*)javaLangString, 16)) {
		clazz->hotFieldOffset =  1; 
		clazz->hotFieldOffset |= J9GCSpecialHotField;
	}
	break;
case 17:
	/* String compare against HashMap & TreeMap only */
case 22:
	/* compare against $Node */
	.....
```

Reply:
Sounds good to me 👍